src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.cs

```diff
@@ -46,46 +61,99 @@ public GtkCoreWindowExtension(object owner)
 			_owner = (CoreWindow)owner;
 			_ownerEvents = (ICoreWindowEvents)owner;
 
-			GtkHost.Window.AddEvents((int)(
-				Gdk.EventMask.PointerMotionMask
-				| EventMask.ButtonPressMask
-				| EventMask.SmoothScrollMask
-			));
-			GtkHost.Window.MotionNotifyEvent += OnWindowMotionEvent;
-			GtkHost.Window.ButtonPressEvent += OnWindowButtonPressEvent;
-			GtkHost.Window.ButtonReleaseEvent += OnWindowButtonReleaseEvent;
+			GtkHost.Window.AddEvents((int)RequestedEvents);
+
 			GtkHost.Window.EnterNotifyEvent += OnWindowEnterEvent;
 			GtkHost.Window.LeaveNotifyEvent += OnWindowLeaveEvent;
+			GtkHost.Window.ButtonPressEvent += OnWindowButtonPressEvent;
+			GtkHost.Window.ButtonReleaseEvent += OnWindowButtonReleaseEvent;
+			GtkHost.Window.MotionNotifyEvent += OnWindowMotionEvent;
 			GtkHost.Window.ScrollEvent += OnWindowScrollEvent;
+			GtkHost.Window.TouchEvent += OnWindowTouchEvent;
+			GtkHost.Window.ProximityInEvent += OnWindowProximityInEvent;
+			GtkHost.Window.ProximityOutEvent += OnWindowProximityOutEvent;
+
+			Console.WriteLine($"[INFO] Version: 2");
 
 			InitializeKeyboard();
 		}
+
 		partial void InitializeKeyboard();
 
-		private static uint GetNextFrameId() => (uint)Interlocked.Increment(ref _currentFrameId);
+		private void OnWindowProximityOutEvent(object o, ProximityOutEventArgs args)
+		{
+			try
+			{
+				Console.Write("PEN_EXIT ");
+				if (AsPointerArgs(args.Event) is { } ptArgs)
+				{
+					_ownerEvents.RaisePointerExited(ptArgs);
+				}
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine($"TOUCH FAILED: {e}");
+			}
+		}
+
+		private void OnWindowProximityInEvent(object o, ProximityInEventArgs args)
+		{
+			try
+			{
+				if (AsPointerArgs(args.Event) is { } ptArgs)
+				{
+					_ownerEvents.RaisePointerEntered(ptArgs);
+				}
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine($"TOUCH FAILED: {e}");
+			}
+		}
+
+		private void OnWindowTouchEvent(object o, TouchEventArgs args)
+		{
+			try
+			{
+				// Note: We DO NOT used the args.Event as it causes an InvalidCastException as of 2021-03-09
```

Comment:
Let's make a PR in the original repo to fix this (and point to the line associated to it in GTK#)